# Length of Stay in a School Project

## Install Libraries

In [1]:
##install the ipython library
!pip install ipython-sql

In [2]:
%load_ext sql



In [3]:
##import libraries 

import pandas as pd
import sqlite3


## Import into the Jupyter first using Python codes 

In [4]:
##import csv file

df = pd.read_csv('students.csv')

In [5]:
#check it

df

inter_dom region  gender academic   age  age_cate  stay stay_cate  \
0       Inter    SEA    Male     Grad  24.0       4.0   5.0      Long   
1       Inter    SEA    Male     Grad  28.0       5.0   1.0     Short   
2       Inter    SEA    Male     Grad  25.0       4.0   6.0      Long   
3       Inter     EA  Female     Grad  29.0       5.0   1.0     Short   
4       Inter     EA  Female     Grad  28.0       5.0   1.0     Short   
..        ...    ...     ...      ...   ...       ...   ...       ...   
281       NaN    NaN     NaN      NaN   NaN       NaN   NaN       NaN   
282       NaN    NaN     NaN      NaN   NaN       NaN   NaN       NaN   
283       NaN    NaN     NaN      NaN   NaN       NaN   NaN       NaN   
284       NaN    NaN     NaN      NaN   NaN       NaN   NaN       NaN   
285       NaN    NaN     NaN      NaN   NaN       NaN   NaN       NaN   

     japanese japanese_cate  ...  friends_bi parents_bi relative_bi  \
0         3.0       Average  ...         Yes        Yes          No   
1         4.0          High  ...         Yes        Yes          No   
2         4.0          High  ...          No         No          No   
3         2.0           Low  ...         Yes        Yes         Yes   
4         1.0           Low  ...         Yes        Yes          No   
..        ...           ...  ...         ...        ...         ...   
281       NaN           NaN  ...         222        NaN         NaN   
282       NaN           NaN  ...         249        NaN         NaN   
283       NaN           NaN  ...         203        NaN         NaN   
284       NaN           NaN  ...         247        NaN         NaN   
285       NaN           NaN  ...         223        NaN         NaN   

    professional_bi phone_bi doctor_bi religion_bi  alone_bi others_bi  \
0                No       No        No          No        No        No   
1                No       No        No          No        No        No   
2                No       No        No          No        No        No   
3               Yes       No        No          No        No        No   
4               Yes       No       Yes         Yes        No        No   
..              ...      ...       ...         ...       ...       ...   
281             NaN      NaN       NaN         NaN       NaN       NaN   
282             NaN      NaN       NaN         NaN       NaN       NaN   
283             NaN      NaN       NaN         NaN       NaN       NaN   
284             NaN      NaN       NaN         NaN       NaN       NaN   
285             NaN      NaN       NaN         NaN       NaN       NaN   

     internet_bi  
0             No  
1             No  
2             No  
3             No  
4             No  
..           ...  
281          NaN  
282          NaN  
283          NaN  
284          NaN  
285          NaN  

[286 rows x 50 columns]

## Connect it with SQL

In [6]:
##now connect to sql

cnn = sqlite3.connect('jupyter_sql_tutorial.db')


In [8]:
##we will add the data to the database by using to_sql
df.to_sql('students', cnn, if_exists='replace')


286

In [9]:
cnn.close()

In [10]:
%sql sqlite:///jupyter_sql_tutorial.db

In [11]:
%%sql

SELECT *
FROM students;

 * sqlite:///jupyter_sql_tutorial.db
Done.


[(0, 'Inter', 'SEA', 'Male', 'Grad', 24.0, 4.0, 5.0, 'Long', 3.0, 'Average', 5.0, 'High', None, 'Yes', 'No', 'No', 'No', 0.0, 'Min', 34.0, 23.0, 9.0, 11.0, 8.0, 11.0, 2.0, 27.0, 91.0, 5.0, 5.0, 6.0, 3.0, 2.0, 1.0, 4.0, 1.0, 3.0, 4.0, None, 'Yes', 'Yes', 'Yes', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No'),
 (1, 'Inter', 'SEA', 'Male', 'Grad', 28.0, 5.0, 1.0, 'Short', 4.0, 'High', 4.0, 'High', None, 'No', 'No', 'No', 'No', 2.0, 'Min', 48.0, 8.0, 7.0, 5.0, 4.0, 3.0, 2.0, 10.0, 39.0, 7.0, 7.0, 7.0, 4.0, 4.0, 4.0, 4.0, 1.0, 1.0, 1.0, None, 'Yes', 'Yes', 'Yes', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No'),
 (2, 'Inter', 'SEA', 'Male', 'Grad', 25.0, 4.0, 6.0, 'Long', 4.0, 'High', 4.0, 'High', 'Yes', 'Yes', 'No', 'No', 'No', 2.0, 'Min', 41.0, 13.0, 4.0, 7.0, 6.0, 4.0, 3.0, 14.0, 51.0, 3.0, 3.0, 3.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, None, 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No'),
 (3, 'Inter', 'EA', 'Female', 'Grad', 29.0, 5.0, 1.0, 'Short', 2.0, 'Low', 3.0, 'Average', 'No', 'No', 'No', 'No', 'No', 3.0, 'Min', 37.0, 16.0, 10.0, 10.0, 8.0, 6.0, 4.0, 21.0, 75.0, 5.0, 5.0, 5.0, 5.0, 5.0, 2.0, 2.0, 2.0, 4.0, 4.0, None, 'Yes', 'Yes', 'Yes', 'Yes', 'Yes', 'No', 'No', 'No', 'No', 'No', 'No'),
 (4, 'Inter', 'EA', 'Female', 'Grad', 28.0, 5.0, 1.0, 'Short', 1.0, 'Low', 3.0, 'Average', 'Yes', 'No', 'No', 'No', 'No', 3.0, 'Min', 37.0, 15.0, 12.0, 5.0, 8.0, 7.0, 4.0, 31.0, 82.0, 5.0, 5.0, 5.0, 2.0, 5.0, 2.0, 5.0, 5.0, 4.0, 4.0, None, 'Yes', 'Yes', 'Yes', 'No', 'Yes', 'No', 'Yes', 'Yes', 'No', 'No', 'No'),
 (5, 'Inter', 'SEA', 'Male', 'Grad', 24.0, 4.0, 6.0, 'Long', 3.0, 'Average', 4.0, 'High', 'Yes', 'No', 'No', 'No', 'No', 6.0, 'Mild', 38.0, 18.0, 8.0, 10.0, 8.0, 7.0, 3.0, 29.0, 83.0, 6.0, 5.0, 4.0, 2.0, 1.0, 1.0, 2.0, 1.0, 5.0, 1.0, None, 'Yes', 'Yes', 'No', 'No', 'No', 'No', 'No', 'No', 'Yes', 'No', 'No'),
 (6, 'Inter', 'SA', 'Male', 'Grad', 23.0, 4.0, 1.0, 'Short', 3.0, 'Average', 5.0, 'High', 'Yes', 'No', 'No', 'No', 'No', 3.0, 'Min', 46.0, 17.0, 6.0, 10.0, 5.0, 3.0, 2.0, 15.0, 58.0, 7.0, 5.0, 7.0, 2.0, 2.0, 1.0, 5.0, 1.0, 1.0, 1.0, None, 'Yes', 'Yes', 'Yes', 'No', 'No', 'No', 'Yes', 'No', 'No', 'No', 'No'),
 (7, 'Inter', 'SEA', 'Female', 'Grad', 30.0, 5.0, 2.0, 'Medium', 1.0, 'Low', 1.0, 'Low', 'Yes', 'Yes', 'Yes', 'No', 'No', 9.0, 'Mild', 41.0, 16.0, 20.0, 19.0, 15.0, 11.0, 6.0, 40.0, 127.0, 7.0, 2.0, 2.0, 2.0, 6.0, 2.0, 1.0, 1.0, 3.0, 1.0, None, 'Yes', 'No', 'No', 'No', 'Yes', 'No', 'No', 'No', 'No', 'No', 'No'),
 (8, 'Inter', 'SEA', 'Female', 'Grad', 25.0, 4.0, 4.0, 'Long', 4.0, 'High', 4.0, 'High', 'No', 'No', 'No', 'Yes', 'Other', 7.0, 'Mild', 36.0, 22.0, 12.0, 13.0, 13.0, 10.0, 6.0, 33.0, 109.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, None, 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No'),
 (9, 'Inter', 'Others', 'Male', 'Grad', 31.0, 5.0, 2.0, 'Medium', 1.0, 'Low', 4.0, 'High', 'Yes', 'Yes', 'No', 'No', 'No', 3.0, 'Min', 48.0, 8.0, 4.0, 5.0, 12.0, 3.0, 2.0, 17.0, 51.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 2.0, 1.0, 2.0, 1.0, None, 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No'),
 (10, 'Inter', 'Others', 'Female', 'Grad', 28.0, 5.0, 1.0, 'Short', 1.0, 'Low', 2.0, 'Low', 'No', 'Yes', 'No', 'No', 'No', 5.0, 'Mild', 32.0, 24.0, 8.0, 10.0, 8.0, 6.0, 6.0, 30.0, 92.0, 7.0, 3.0, 7.0, 7.0, 3.0, 1.0, 3.0, 1.0, 3.0, 5.0, None, 'Yes', 'No', 'Yes', 'Yes', 'No', 'No', 'No', 'No', 'No', 'Yes', 'No'),
 (11, 'Inter', 'SEA', 'Female', 'Grad', 31.0, 5.0, 1.0, 'Short', 1.0, 'Low', 4.0, 'High', 'Yes', 'Yes', 'No', 'No', 'No', 8.0, 'Mild', 47.0, 17.0, 12.0, 14.0, 13.0, 9.0, 4.0, 26.0, 95.0, 6.0, 2.0, 6.0, 2.0, 2.0, 2.0, 2.0, 4.0, 2.0, 5.0, None, 'Yes', 'No', 'Yes', 'No', 'No', 'No', 'No', 'No', 'No', 'Yes', 'No'),
 (12, 'Inter', 'SA', 'Male', 'Grad', 29.0, 5.0, 1.0, 'Short', 1.0, 'Low', 4.0, 'High', 'Yes', 'Yes', 'No', 'No', 'No', 1.0, 'Min', 48.0, 8.0, 11.0, 5.0, 4.0, 7.0, 2.0, 17.0, 54.0, 7.0, 5.0, 7.0, 7.0, 1.0, 1.0, 1.0, 4.0, 2.0, 1.0, None, 'Yes', 'Yes', 'Yes', 'Yes', 'No', 'No', 'N

In [ ]:
##Explore and analyze the students data to see how the length of stay (stay) impacts the average mental health diagnostic 
scores of the international students present in the study.

Field Name	Description
inter_dom	Types of students (international or domestic)
japanese_cate	Japanese language proficiency
english_cate	English language proficiency
academic	Current academic level (undergraduate or graduate)
age	Current age of student
stay	Current length of stay in years
todep	Total score of depression (PHQ-9 test)
tosc	Total score of social connectedness (SCS test)
toas	Total score of acculturative stress (ASISS test)

In [22]:
%%sql

SELECT stay, 
    COUNT(*) AS count_int,
    ROUND(AVG(todep), 2) AS average_phq, 
    ROUND(AVG(tosc), 2) AS average_scs, 
    ROUND(AVG(toas), 2) AS average_as,
    (ROUND(AVG(todep), 2) + ROUND(AVG(tosc), 2) + ROUND(AVG(toas), 2)) AS total_average_score
FROM students
WHERE inter_dom = 'Inter'
GROUP BY stay
ORDER BY stay DESC;
     

 * sqlite:///jupyter_sql_tutorial.db
Done.


[(10.0, 1, 13.0, 32.0, 50.0, 95.0),
 (8.0, 1, 10.0, 44.0, 65.0, 119.0),
 (7.0, 1, 4.0, 48.0, 45.0, 97.0),
 (6.0, 3, 6.0, 38.0, 58.67, 102.67),
 (5.0, 1, 0.0, 34.0, 91.0, 125.0),
 (4.0, 14, 8.57, 33.93, 87.71, 130.20999999999998),
 (3.0, 46, 9.09, 37.13, 78.0, 124.22),
 (2.0, 39, 8.28, 37.08, 77.67, 123.03),
 (1.0, 95, 7.48, 38.11, 72.8, 118.39)]

##Analysis

Depression (average_phq):
Depression scores fluctuate across different lengths of stay. The highest depression score is 13 for a 10-year stay, while the lowest is 0 for a 5-year stay.
Individuals with shorter stays (1 year) have a moderately low depression score (7.48), while those with a stay of 4 to 6 years seem to experience milder depression.

Social Connectedness (average_scs):
Social connectedness generally increases as the length of stay decreases. For example, individuals with a 1-year stay report an average score of 38.11, while those staying for 10 years report a lower score of 32.0.
There is a notable peak in social connectedness for individuals staying 7 and 8 years, with scores of 48.0 and 44.0, respectively.

Acculturative Stress (average_as):
Acculturative stress increases as the length of stay increases, peaking at 5 years (91.0). This could suggest that stress related to cultural adaptation builds up over time and is highest after a mid-length stay.
For shorter stays (1 to 3 years), acculturative stress levels are slightly lower, around the mid-70s range.
Total Score (Total_score):

The total score, which is the sum of the three measures (PHQ-9, SCS, and ASISS), generally increases with the length of stay, peaking at 130.21 for individuals staying 4 years and remaining high for those with 3- to 5-year stays.
The highest total score is seen in the 4- to 5-year range, indicating that individuals staying for this period experience a combination of depression, social connectedness, and acculturative stress at higher levels compared to others.

##Summary

According to the research, people tend to become less socially engaged and gradually experience more acculturative stress 
the longer they stay, with a notable peak occurring around the 5-year mark. Depression scores vary, declining after eight 
years and then increasing once more as the stay shortens to three years or less. Shorter stays—a year, 
for example—are associated with stronger levels of social connectivity and lower levels of acculturative stress. 
This suggests that longer stays may provide more difficulties for social bonds and environment adaptation.